### PR TITLE
Add .NET content in guides

### PR DESCRIPTION
**Pull request by laura-rodriguez**
## Description:
- **What's changed?**
Added content for .NET into the "Sign in into your app" guide.
- **Is this PR related to a Monolith release?** 
No

### Resolves:

* [OKTA-214013](https://oktainc.atlassian.net/browse/OKTA-214013)

### CODE REVIEW

**Comment by laura-rodriguez** [comment]
> you should set this to merge into the `guides` branch until we are ready to release Guides

Doh!  @swiftone you're right.

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/add-signin-button/index.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: Add a Button to Sign In
+---
+# Add a Button to Sign In
+
+In your application, you want to provide a button to sign the user in. When the user clicks this button, they are redirected to the Okta-hosted sign-in page so they can authenticate.
+
+<StackSelector snippet="login-redirect"/>
+
+<!-- <NextSectionLink/> -->
```
Comment (swiftone) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/add-signin-button/index.md]: If you grab the latest `guides` the NextSectionLink should work

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/add-signin-button/index.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+title: Add a Button to Sign In
+---
+In your application, you want to provide a button to sign the user in. When the user clicks this button, they are redirected to the Okta-hosted sign-in page so they can authenticate.
```
Comment (jakubvul) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/add-signin-button/index.md]: I'd remove that comma: `In your application you want to`

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/before-you-begin/index.md [hunk 3]
```diff
@@ -11,18 +12,12 @@ This guide walks you through how to sign users into your existing web app. It is
 ## Before You Begin
 These instructions assume that you: 
 
-* Are building Web app that's rendered by a server, otherwise see [Sign Users into a Single-Page App](sign-into-spa). If you are building a mobile app, see [Sign Users into a Mobile App](sign-into-mobile-app). If you are building an API, see [Protect an API].
+* Are building a web app that's rendered by a server, otherwise see [Sign Users into a Single-Page App](sign-into-spa). If you are building a mobile app, see [Sign Users into a Mobile App](sign-into-mobile-app). If you are building an API, see [Protect an API].
```
Comment (jakubvul) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/before-you-begin/index.md]: Can we make this into a two-level list?


* Are building a web app that's rendered by a server. 
  * If you are building a web app that's rendered by a client, see [Sign Users into a Single-Page App](sign-into-spa). 
  * If you are building a mobile app, see [Sign Users into a Mobile App](sign-into-mobile-app). 
  * If you are building an API, see [Protect an API].

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/net/configmid.md [hunk 4]
```diff
@@ -1 +1,60 @@
...
+
+        <!-- 2. Update the Okta application with these values -->
+        <add key="okta:RedirectUri" value="http://localhost:8080/authorization-code/callback" />
+        <add key="okta:PostLogoutRedirectUri" value="http://localhost:8080/Account/PostLogout" />
+    </appSettings>
+    ...
+  </configuration>
+  ```
+
+Now is time to add an OWIN Startup class to your project where you will have to configure the Okta middleware. Right-click your project and select **Add > OWIN Startup Class**. Name the file `Startup.cs`.
```
Comment (jakubvul) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/net/configmid.md]: Now **it's** time

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/net/configmid.md [hunk 5]
```diff
@@ -1 +1,60 @@
...
+using System.Collections.Generic;
+using System.Configuration;
+using Microsoft.Owin;
+using Microsoft.Owin.Security;
+using Microsoft.Owin.Security.Cookies;
+using Okta.AspNet;
+using Owin;
+```
+
+*Note:* If you are using .NET framework <4.6 or you are getting the following error: The request was aborted: Could not create SSL/TLS secure channel. Make sure to include the following code in the Application_Start or Startup:
```
Comment (jakubvul) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/net/configmid.md]: Probably should put quotations around the error, just to be clear where the error text begins and ends.

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/define-callback-route/index.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+1. Your application directs the browser to the Okta Sign-In page, where the user authenticates.
+1. The browser receives an authorization code from your Okta authorization server.
+1. The authorization code is passed to your application.
+1. Your application sends this code to Okta, and Okta returns access and ID tokens, and optionally a refresh token.
+1. Your application can now use these tokens to call the resource server (for example an API) on behalf of the user.
+
+### Login Redirect URI
+
+In your web app, you should define a redirect uri which is the callback location where the user-agent will be directed to along with the code. This must match one of the **Login redirect URIs** you specified when you create your Okta application.
```
Comment (jakubvul) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/define-callback-route/index.md]: `define a redirect URI`

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/require-authentication/net/reqautheverything.md [hunk 7]
```diff
@@ -1 +1,24 @@
...
+{
+    public static void RegisterGlobalFilters(GlobalFilterCollection filters)
+    {
+        ...
+        filters.Add(new AuthorizeAttribute());
+    }
+}
+```
+
+Also, you can opt to create a `BaseProtectedController`and make all your controllers to inherits it:
```
Comment (jakubvul) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/require-authentication/net/reqautheverything.md]: `and make all your controllers inherit it:`

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/require-authentication/netcore/reqautheverything.md [hunk 8]
```diff
@@ -1 +1,26 @@
...
+    {
+        var policy = new AuthorizationPolicyBuilder()
+            .RequireAuthenticatedUser()
+            .Build();
+        o.Filters.Add(new AuthorizeFilter(policy));
+    });
+}
+```
+
+Also, you can opt to create a `BaseProtectedController` decorated with the `Authorize` attribute and make all your controllers to inherits it:
```
Comment (jakubvul) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/require-authentication/netcore/reqautheverything.md]: `and make all your controllers inherit it:`

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/before-you-begin/index.md (hunk 3 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/guides/sign-into-your-web-app/before-you-begin/index.md]: These are actually commented out right now. So, we're going to add them back in later when all of the guides are done.

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/net/configmid.md (hunk 4 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/net/configmid.md]: or: 

Now add an OWIN Startup....

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/net/configmid.md (hunk 5 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/net/configmid.md]: done

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/define-callback-route/index.md (hunk 6 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/guides/sign-into-your-web-app/define-callback-route/index.md]: done

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/require-authentication/net/reqautheverything.md (hunk 7 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/guides/sign-into-your-web-app/require-authentication/net/reqautheverything.md]: done

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/require-authentication/netcore/reqautheverything.md (hunk 8 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/guides/sign-into-your-web-app/require-authentication/netcore/reqautheverything.md]: Done

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/add-signin-button/index.md (hunk 2 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/guides/sign-into-your-web-app/add-signin-button/index.md]: done

**Comment by nbarbettini** [comment]
@brettritter-okta There does seem to be something exploding on the Before You Begin step:

image

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/before-you-begin/index.md [hunk 9]
```diff
@@ -1,22 +1,19 @@
...
-* .NET (ASP.NET)
-* .NET (ASP.NET Core)
-
-## Before You Begin
-These instructions assume that you: 
-
-* Are building a web app that's rendered by a server, otherwise see [Sign Users into a Single-Page App](sign-into-spa). If you are building a mobile app, see [Sign Users into a Mobile App](sign-into-mobile-app). If you are building an API, see [Protect an API].
-* Have an Okta Developer organization. If you don't, you can get one here: <https://developer.okta.com/signup>
+* Have an Okta Developer Edition organization. (Don't have one? [Create one for free](https://developer.okta.com/signup).)
+* Know the basics of building JavaScript applications.
```
Comment (laura-rodriguez) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/before-you-begin/index.md]: @susanharper-okta is JavaScript correct here?

Comment (susanharper-okta) [reply packages/@okta/vuepress-site/guides/sign-into-your-web-app/before-you-begin/index.md]: uh nope....that one can be removed......weird - not even sure where that came from...

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/next-steps/net/samples.md [hunk 10]
```diff
@@ -0,0 +1 @@
+We have samples available in our [ASP.NET MVC Sample repo](https://github.com/okta/samples-aspnet/tree/master/okta-hosted-login) on GitHub.
```
Comment (laura-rodriguez) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/next-steps/net/samples.md]: do we want to point to the `okta-hosted-login` samples or the samples root folder? @nbarbettini @susanharper-okta

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/before-you-begin/index.md (hunk 9 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/guides/sign-into-your-web-app/before-you-begin/index.md]: Yup! Perfect! Thank you.

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/aspnet/addconfigpkg.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+Install the following dependencies in your project:
+
+* [Microsoft.Owin.Security.Cookies](https://www.nuget.org/packages/Microsoft.Owin.Security.Cookies) 4.0.0 or higher 
+* [Microsoft.Owin.Host.SystemWeb](https://www.nuget.org/packages/Microsoft.Owin.Host.SystemWeb) 4.0.0 or higher
+* [Okta.AspNet](https://www.nuget.org/packages/Okta.AspNet)
+
+You can install them via the **NuGet Package Manager**:
+
+1. In the Solution Explorer, right-click on your solution and select **Manage Nuget Packages for Solution**.
+1. Click **Browse** and search for the package that you want to install, for example, `Okta.AspNet`.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/aspnet/addconfigpkg.md]: the numbering here needs to be fixed.

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/aspnetcore/addconfigpkg.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+
+Install the following dependencies in your project:
+
+* [Microsoft.AspNetCore.Authentication.Cookies](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Cookies/) 2.2.0 or higher 
+* [Okta.AspNetCore](https://www.nuget.org/packages/Okta.AspNetCore)
+
+You can install them via the **NuGet Package Manager**:
+
+1. In the Solution Explorer, right-click on your solution and select **Manage Nuget Packages for Solution**.
+1. Click **Browse** and search for the package you want to install, for example, `Okta.AspNetCore`.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/aspnetcore/addconfigpkg.md]: The numbering here needs to be fixed.

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/define-callback-route/index.md [hunk 13]
```diff
@@ -0,0 +1,27 @@
+---
+title: Define a Callback Route
+---
+
+If you are building a server-side (or "web") application that is capable of securely storing secrets, then the authorization code flow is the recommended method for controlling access to it.
+
+At a high-level, this flow has the following steps:
+
+1. Your application directs the browser to the Okta Sign-In page, where the user authenticates.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/define-callback-route/index.md]: Numbering needs to be fixed.

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/aspnet/addconfigpkg.md (hunk 11 above)
Comment (laura-rodriguez) [reply packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/aspnet/addconfigpkg.md]: > the numbering here needs to be fixed.

For ordered lists, you can just declare (1.) and VuePress will do the magic for you :)
https://vuepressbook.com/tutorial/tutorial4.html#markdown-lists

Comment (susanharper-okta) [reply packages/@okta/vuepress-site/guides/sign-into-your-web-app/configure-packages/aspnet/addconfigpkg.md]: d'oh! Thank you! I just built it locally and all is good. Should've done that first. My bad. Thanks!

File: packages/@okta/vuepress-site/guides/sign-into-your-web-app/get-user-info/index.md [hunk 14]
```diff
@@ -1,12 +1,15 @@
...
+When a user signs in, their profile information (stored in Okta) is made available to your application. It's common to use this info to update your application's UI.
+
+By default, the profile items (called "claims") that Okta returns include the user's email address, first name, and last name. 
 
 <StackSelector snippet="getuserinfo"/>
 
-### Customize the Claims Returned
+<!-- You can also customize the items (called claims) that are returned from Okta. See [Token customization guide]. -->
+
+<NextSectionLink>All done!</NextSectionLink>
```
Comment (swiftone) [inline packages/@okta/vuepress-site/guides/sign-into-your-web-app/get-user-info/index.md]: Do we want to have this on the last page?

**Pull request merged**